### PR TITLE
Fix gcc-14 C++26 nightly jenkins build

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://github.com/kokkos/kokkos/pull/7081#issuecomment-2227543500 and addresses the warnings in that build.
- We need to install a recent CMake version and it was easiest to just install CMake 3.30 by hand.
- one of our guards only addressed C++23 but not C++26
- g++-14 complained about uninitialized variables (likely spurious):
```
In file included from /app/kokkos/simd/unit_tests/TestSIMD.cpp:20:
In function 'void host_check_shift_by_lanes_on_one_loader(ShiftOp, DataType*, Kokkos::Experimental::simd<T, Abi>&) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; Loader = load_masked; ShiftOp = shift_right; DataType = long unsigned int]',
    inlined from 'void host_check_shift_op_all_loaders(ShiftOp, DataType*, DataType*, std::size_t) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; ShiftOp = shift_right; DataType = long unsigned int]' at /app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:96:60,
    inlined from 'void host_check_shift_ops() [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = long unsigned int]' at /app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:123:41:
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:68:14: error: 'simd_vals' may be used uninitialized [-Werror=maybe-uninitialized]
   68 |     DataType value = simd_vals[lane];
      |              ^~~~~
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp: In function 'void host_check_shift_ops() [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = long unsigned int]':
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:61:13: note: 'simd_vals' declared here
   61 |   simd_type simd_vals;
      |             ^~~~~~~~~
In function 'void host_check_shift_by_lanes_on_one_loader(ShiftOp, DataType*, Kokkos::Experimental::simd<T, Abi>&) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; Loader = load_masked; ShiftOp = shift_left; DataType = long unsigned int]',
    inlined from 'void host_check_shift_op_all_loaders(ShiftOp, DataType*, DataType*, std::size_t) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; ShiftOp = shift_left; DataType = long unsigned int' at /app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:96:60,
    inlined from 'void host_check_shift_ops() [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = long unsigned int]' at /app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:125:41:
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:68:14: error: 'simd_vals' may be used uninitialized [-Werror=maybe-uninitialized]
   68 |     DataType value = simd_vals[lane];
      |              ^~~~~
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp: In function 'void host_check_shift_ops() [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = long unsigned int]':
/app/kokkos/simd/unit_tests/include/TestSIMD_ShiftOps.hpp:61:13: note: 'simd_vals' declared here
   61 |   simd_type simd_vals;
      |             ^~~~~~~~~
cc1plus: all warnings being treated as errors
```

### CODE REVIEW

File: simd/unit_tests/include/SIMDTesting_Utilities.hpp [hunk 1]
```diff
@@ -135,8 +135,8 @@ class load_masked {
     for (std::size_t i = 0; i < n; ++i) {
       mask[i] = true;
     }
+    result = T(0);
     where(mask, result).copy_from(mem, Kokkos::Experimental::simd_flag_default);
-    where(!mask, result) = 0;
     return true;
```
Comment (masterleinad) [inline simd/unit_tests/include/SIMDTesting_Utilities.hpp:140]: I'm not overly happy with this, but I couldn't stop the compiler from complaining differently.

Comment (masterleinad) [reply simd/unit_tests/include/SIMDTesting_Utilities.hpp:140]: @ldh4 Any better idea?

Comment (ldh4) [reply simd/unit_tests/include/SIMDTesting_Utilities.hpp:140]: I think the change is fine, but I find it interesting that this loader is the only one getting flagged. Other loaders are also not explicitly initializing the value of `result` either.

**Review by ldh4 — APPROVED** [review]

File: .jenkins_nightly [hunk 2]
```diff
@@ -80,11 +80,12 @@ pipeline {
                     steps {
                         sh '''
                           DEBIAN_FRONTEND=noninteractive && \
-                          apt-get update && apt-get upgrade -y && apt-get install -y \
-                          cmake \
-                          && \
+                          apt-get update && apt-get upgrade -y && \
```
Comment (aprokop) [inline .jenkins_nightly:83]: Why do you need to do any of these if you are not installing anything?

Comment (masterleinad) [reply .jenkins_nightly:83]: This worked. We can try without it.

Comment (masterleinad) [reply .jenkins_nightly]: Seems to be fine without it. Updated.

Comment (dalg24) [reply .jenkins_nightly]: You run everything wight not just building the image?

Comment (masterleinad) [reply .jenkins_nightly]: Yes, I configured the same way we specify here, build and ran the tests.

**Pull request merged**